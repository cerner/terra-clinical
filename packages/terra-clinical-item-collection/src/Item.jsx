import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import ItemView from 'terra-clinical-item-view';
import List from 'terra-list';
import Table from 'terra-table';
import 'terra-base/lib/baseStyles';
import styles from './ItemCollection.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The display elements to be presented.
   */
  children: PropTypes.node,
  /**
   * The start accessory element to be presented.
   * This will only be shown if the Item Collection component's `hasStartAccessory` prop is *true*.
   */
  startAccessory: PropTypes.node,
  /**
   * The comment element to be presented.
   * This will only be shown if the Item Collection component's `hasComment` prop is *true*.
   */
  comment: PropTypes.node,
  /**
   * The end accessory element to be presented.
   * This will only be shown if the Item Collection component's `hasEndAccessory` prop is *true*.
   */
  endAccessory: PropTypes.node,
  /**
   * When displayed as a list item, the column layout in which to present the displays.
   * One of `'oneColumn'`, `'twoColumns'`.
   */
  listItemLayout: PropTypes.oneOf(['oneColumn', 'twoColumns']),
  /**
   * When displayed as a list item, the text color emphasis when using the two columns layout.
   * One of `'default'`, `'start'`.
   */
  listItemTextEmphasis: PropTypes.oneOf(['default', 'start']),
  /**
   * When displayed as a list item, whether or not all text should truncate.
   */
  isListItemTruncated: PropTypes.bool,
  /**
   * The vertical alignment of the start and end accessories. One of `'alignTop'`,`'alignCenter'`.
   */
  accessoryAlignment: PropTypes.oneOf(['alignTop', 'alignCenter']),
  /**
   * When displayed as a list item, indicates whether or not space is allocated for the start accessory.
   * This will be set by the Item Collection component.
   */
  reserveStartAccessorySpace: PropTypes.bool,
  /**
   * When displayed as a list item, whether or not it has a chevron. This should only be used when creating a
   * a single selectable list.
   */
  showListItemChevron: PropTypes.bool,
  /**
   * Whether or not the item is selectable. If true, the item is given list and table hover and focus styles
   * and set tabIndex to 0.
   */
  isSelectable: PropTypes.bool,
  /*
   * Whether or not the item is selected.
   */
  isSelected: PropTypes.bool,
  /**
   * The view in which the item should be presented. Options are list or table and this will be set by the
   * Item Collection component.
   */
  view: PropTypes.oneOf(['list', 'table']),
};

const defaultProps = {
  startAccessory: undefined,
  comment: undefined,
  endAccessory: undefined,
  accessoryAlignment: 'alignCenter',
  isListItemTruncated: false,
  isSelectable: false,
  isSelected: false,
  listItemLayout: 'oneColumn',
  listItemTextEmphasis: 'default',
  reserveStartAccessorySpace: false,
  showListItemChevron: false,
  view: 'list',
};

function createListItem(elements, selectableProps, customProps, isSelected, itemViewStyles, showListItemChevron) {
  const {
    startAccessory, children, comment, endAccessory, reserveStartAccessorySpace,
  } = elements;

  const listItemContent = (
    <ItemView
      startAccessory={startAccessory}
      displays={React.Children.toArray(children)}
      comment={comment}
      endAccessory={endAccessory}
      reserveStartAccessorySpace={reserveStartAccessorySpace}
      {...itemViewStyles}
    />
  );

  return (
    <List.Item
      content={listItemContent}
      isSelected={isSelected}
      hasChevron={showListItemChevron}
      {...selectableProps}
      {...customProps}
    />
  );
}

function createTableCell(content, keyValue, contentType, accessoryAlignment) {
  const cellClassNames = cx(
    `content-${contentType}`,
    { 'content-end-accessory': keyValue === 'end_accessory' },
    { 'content-accessory-align-center': (contentType.includes('accessory') && accessoryAlignment === 'alignCenter') },
    { 'content-accessory-align-top': (contentType.includes('accessory') && accessoryAlignment === 'alignTop') },
  );

  return (<Table.Cell content={content} key={keyValue} className={cellClassNames} />);
}

function createTableRow(elements, selectableProps, customProps, isSelected, accessoryAlignment) {
  const {
    startAccessory, children, comment, endAccessory,
  } = elements;

  const displayContent = React.Children.map(children, (display, index) => {
    const displayKey = `display_${index + 1}`;
    return createTableCell(display, displayKey, 'display');
  });

  return (
    <Table.Row isSelected={isSelected} {...selectableProps} {...customProps}>
      {startAccessory && createTableCell(startAccessory, 'start_accessory', 'accessory', accessoryAlignment)}
      {displayContent}
      {comment && createTableCell(comment, 'comment', 'comment')}
      {endAccessory && createTableCell(endAccessory, 'end_accessory', 'accessory', accessoryAlignment)}
    </Table.Row>
  );
}

const Item = (props) => {
  const {
    children,
    startAccessory,
    comment,
    endAccessory,
    accessoryAlignment,
    isListItemTruncated,
    isSelectable,
    isSelected,
    listItemLayout,
    listItemTextEmphasis,
    reserveStartAccessorySpace,
    showListItemChevron,
    view,
    ...customProps
  } = props;

  const elements = {
    children, startAccessory, comment, endAccessory, reserveStartAccessorySpace,
  };
  const selectableProps = isSelectable ? { isSelectable, tabIndex: 0 } : {};

  if (view === 'table') {
    return createTableRow(elements, selectableProps, customProps, isSelected, accessoryAlignment);
  }

  const itemViewStyles = {
    layout: listItemLayout, textEmphasis: listItemTextEmphasis, isTruncated: isListItemTruncated, accessoryAlignment,
  };
  return createListItem(elements, selectableProps, customProps, isSelected, itemViewStyles, showListItemChevron);
};

Item.propTypes = propTypes;
Item.defaultProps = defaultProps;
Item.Display = ItemView.Display;
Item.Comment = ItemView.Comment;

export default Item;
