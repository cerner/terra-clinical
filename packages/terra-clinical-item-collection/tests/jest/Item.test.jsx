import React from 'react';
import Item from '../../src/Item';

const listItemStyles = {
  listItemLayout: 'twoColumns',
  listItemTextEmphasis: 'start',
  isListItemTruncated: true,
  accessoryAlignment: 'alignTop',
};

it('should render a default Item', () => {
  const item = (<Item />);
  const component = shallow(item);
  expect(component).toMatchSnapshot();
});

describe('List View Tests', () => {
  it('should render a default Item', () => {
    const item = (
      <Item view="list" />
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with a start accessory', () => {
    const item = (
      <Item
        view="list"
        startAccessory={<p>start</p>}
      />
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with children', () => {
    const item = (
      <Item view="list">
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with a single children', () => {
    const item = (
      <Item view="list">
        <Item.Display text="Display 1" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with a comment', () => {
    const item = (
      <Item
        view="list"
        comment={<Item.Comment text="comment" />}
      />
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with an end accessory', () => {
    const item = (
      <Item
        view="list"
        endAccessory={<p>end</p>}
      />
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with all elements', () => {
    const item = (
      <Item
        view="list"
        startAccessory={<p>start</p>}
        comment={<Item.Comment text="comment" />}
        endAccessory={<p>end</p>}
      >
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a selectable Item', () => {
    const item = (
      <Item
        view="list"
        startAccessory={<p>start</p>}
        comment={<Item.Comment text="comment" />}
        endAccessory={<p>end</p>}
        isSelectable
      >
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a selected Item', () => {
    const item = (
      <Item
        view="table"
        startAccessory={<p>start</p>}
        comment={<Item.Comment text="comment" />}
        endAccessory={<p>end</p>}
        isSelected
      >
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with list item styles', () => {
    const item = (
      <Item
        view="list"
        startAccessory={<p>start</p>}
        comment={<Item.Comment text="comment" />}
        endAccessory={<p>end</p>}
        {...listItemStyles}
      >
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should pass customProps to the Item', () => {
    const item = (
      <Item
        view="list"
        className="ListItem"
      >
        <Item.Display text="Display 1" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });
});

describe('Table View Tests', () => {
  it('should render a default Item', () => {
    const item = (
      <Item view="table" />
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with a start accessory', () => {
    const item = (
      <Item
        view="table"
        startAccessory={<p>start</p>}
      />
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with children', () => {
    const item = (
      <Item view="table">
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with a single children', () => {
    const item = (
      <Item view="table">
        <Item.Display text="Display 1" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with a comment', () => {
    const item = (
      <Item
        view="table"
        comment={<Item.Comment text="comment" />}
      />
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with an end accessory', () => {
    const item = (
      <Item
        view="table"
        endAccessory={<p>end</p>}
      />
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item with all elements', () => {
    const item = (
      <Item
        view="table"
        startAccessory={<p>start</p>}
        comment={<Item.Comment text="comment" />}
        endAccessory={<p>end</p>}
      >
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a selectable Item', () => {
    const item = (
      <Item
        view="table"
        startAccessory={<p>start</p>}
        comment={<Item.Comment text="comment" />}
        endAccessory={<p>end</p>}
        isSelectable
      >
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a selected Item', () => {
    const item = (
      <Item
        view="table"
        startAccessory={<p>start</p>}
        comment={<Item.Comment text="comment" />}
        endAccessory={<p>end</p>}
        isSelected
      >
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should render a Item that ignores list item styles', () => {
    const item = (
      <Item
        view="table"
        startAccessory={<p>start</p>}
        comment={<Item.Comment text="comment" />}
        endAccessory={<p>end</p>}
        {...listItemStyles}
      >
        <Item.Display text="Display 1" />
        <Item.Display text="Display 2" />
        <Item.Display text="Display 3" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });

  it('should pass customProps to the Item', () => {
    const item = (
      <Item
        view="table"
        className="TableRow"
      >
        <Item.Display text="Display 1" />
      </Item>
    );
    const component = shallow(item);
    expect(component).toMatchSnapshot();
  });
});
