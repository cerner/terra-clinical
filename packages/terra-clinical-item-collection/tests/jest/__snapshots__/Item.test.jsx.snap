// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List View Tests should pass customProps to the Item 1`] = `
<ListItem
  className="ListItem"
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      displays={
        Array [
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 1"
          />,
        ]
      }
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a Item with a comment 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      comment={
        <ItemComment
          isTruncated={false}
          text="comment"
        />
      }
      displays={Array []}
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a Item with a list item chevron 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      comment={
        <ItemComment
          isTruncated={false}
          text="comment"
        />
      }
      displays={
        Array [
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 1"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 2"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 3"
          />,
        ]
      }
      endAccessory={
        <p>
          end
        </p>
      }
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      startAccessory={
        <p>
          start
        </p>
      }
      textEmphasis="default"
    />
  }
  hasChevron={true}
  isSelected={false}
/>
`;

exports[`List View Tests should render a Item with a single children 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      displays={
        Array [
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 1"
          />,
        ]
      }
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a Item with a start accessory 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      displays={Array []}
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      startAccessory={
        <p>
          start
        </p>
      }
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a Item with all elements 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      comment={
        <ItemComment
          isTruncated={false}
          text="comment"
        />
      }
      displays={
        Array [
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 1"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 2"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 3"
          />,
        ]
      }
      endAccessory={
        <p>
          end
        </p>
      }
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      startAccessory={
        <p>
          start
        </p>
      }
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a Item with an end accessory 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      displays={Array []}
      endAccessory={
        <p>
          end
        </p>
      }
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a Item with children 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      displays={
        Array [
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 1"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 2"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 3"
          />,
        ]
      }
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a Item with list item styles 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignTop"
      comment={
        <ItemComment
          isTruncated={false}
          text="comment"
        />
      }
      displays={
        Array [
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 1"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 2"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 3"
          />,
        ]
      }
      endAccessory={
        <p>
          end
        </p>
      }
      isTruncated={true}
      layout="twoColumns"
      reserveStartAccessorySpace={false}
      startAccessory={
        <p>
          start
        </p>
      }
      textEmphasis="start"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a default Item 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      displays={Array []}
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;

exports[`List View Tests should render a selectable Item 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      comment={
        <ItemComment
          isTruncated={false}
          text="comment"
        />
      }
      displays={
        Array [
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 1"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 2"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 3"
          />,
        ]
      }
      endAccessory={
        <p>
          end
        </p>
      }
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      startAccessory={
        <p>
          start
        </p>
      }
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelectable={true}
  isSelected={false}
  tabIndex={0}
/>
`;

exports[`List View Tests should render a selected Item 1`] = `
<TableRow
  isSelected={true}
>
  <TableCell
    className="content-accessory content-accessory-align-center"
    content={
      <p>
        start
      </p>
    }
    key="start_accessory"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 1"
      />
    }
    key="display_1/.0"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 2"
      />
    }
    key="display_2/.1"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 3"
      />
    }
    key="display_3/.2"
  />
  <TableCell
    className="content-comment"
    content={
      <ItemComment
        isTruncated={false}
        text="comment"
      />
    }
    key="comment"
  />
  <TableCell
    className="content-accessory content-end-accessory content-accessory-align-center"
    content={
      <p>
        end
      </p>
    }
    key="end_accessory"
  />
</TableRow>
`;

exports[`Table View Tests should pass customProps to the Item 1`] = `
<TableRow
  className="TableRow"
  isSelected={false}
>
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 1"
      />
    }
    key="display_1/.0"
  />
</TableRow>
`;

exports[`Table View Tests should render a Item that ignores list item styles 1`] = `
<TableRow
  isSelected={false}
>
  <TableCell
    className="content-accessory content-accessory-align-top"
    content={
      <p>
        start
      </p>
    }
    key="start_accessory"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 1"
      />
    }
    key="display_1/.0"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 2"
      />
    }
    key="display_2/.1"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 3"
      />
    }
    key="display_3/.2"
  />
  <TableCell
    className="content-comment"
    content={
      <ItemComment
        isTruncated={false}
        text="comment"
      />
    }
    key="comment"
  />
  <TableCell
    className="content-accessory content-end-accessory content-accessory-align-top"
    content={
      <p>
        end
      </p>
    }
    key="end_accessory"
  />
</TableRow>
`;

exports[`Table View Tests should render a Item that ignores the list item chevron 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      comment={
        <ItemComment
          isTruncated={false}
          text="comment"
        />
      }
      displays={
        Array [
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 1"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 2"
          />,
          <ItemDisplay
            iconAlignment="center"
            isDisabled={false}
            isTruncated={false}
            text="Display 3"
          />,
        ]
      }
      endAccessory={
        <p>
          end
        </p>
      }
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      startAccessory={
        <p>
          start
        </p>
      }
      textEmphasis="default"
    />
  }
  hasChevron={true}
  isSelected={false}
/>
`;

exports[`Table View Tests should render a Item with a comment 1`] = `
<TableRow
  isSelected={false}
>
  <TableCell
    className="content-comment"
    content={
      <ItemComment
        isTruncated={false}
        text="comment"
      />
    }
    key="comment"
  />
</TableRow>
`;

exports[`Table View Tests should render a Item with a single children 1`] = `
<TableRow
  isSelected={false}
>
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 1"
      />
    }
    key="display_1/.0"
  />
</TableRow>
`;

exports[`Table View Tests should render a Item with a start accessory 1`] = `
<TableRow
  isSelected={false}
>
  <TableCell
    className="content-accessory content-accessory-align-center"
    content={
      <p>
        start
      </p>
    }
    key="start_accessory"
  />
</TableRow>
`;

exports[`Table View Tests should render a Item with all elements 1`] = `
<TableRow
  isSelected={false}
>
  <TableCell
    className="content-accessory content-accessory-align-center"
    content={
      <p>
        start
      </p>
    }
    key="start_accessory"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 1"
      />
    }
    key="display_1/.0"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 2"
      />
    }
    key="display_2/.1"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 3"
      />
    }
    key="display_3/.2"
  />
  <TableCell
    className="content-comment"
    content={
      <ItemComment
        isTruncated={false}
        text="comment"
      />
    }
    key="comment"
  />
  <TableCell
    className="content-accessory content-end-accessory content-accessory-align-center"
    content={
      <p>
        end
      </p>
    }
    key="end_accessory"
  />
</TableRow>
`;

exports[`Table View Tests should render a Item with an end accessory 1`] = `
<TableRow
  isSelected={false}
>
  <TableCell
    className="content-accessory content-end-accessory content-accessory-align-center"
    content={
      <p>
        end
      </p>
    }
    key="end_accessory"
  />
</TableRow>
`;

exports[`Table View Tests should render a Item with children 1`] = `
<TableRow
  isSelected={false}
>
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 1"
      />
    }
    key="display_1/.0"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 2"
      />
    }
    key="display_2/.1"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 3"
      />
    }
    key="display_3/.2"
  />
</TableRow>
`;

exports[`Table View Tests should render a default Item 1`] = `
<TableRow
  isSelected={false}
/>
`;

exports[`Table View Tests should render a selectable Item 1`] = `
<TableRow
  isSelectable={true}
  isSelected={false}
  tabIndex={0}
>
  <TableCell
    className="content-accessory content-accessory-align-center"
    content={
      <p>
        start
      </p>
    }
    key="start_accessory"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 1"
      />
    }
    key="display_1/.0"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 2"
      />
    }
    key="display_2/.1"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 3"
      />
    }
    key="display_3/.2"
  />
  <TableCell
    className="content-comment"
    content={
      <ItemComment
        isTruncated={false}
        text="comment"
      />
    }
    key="comment"
  />
  <TableCell
    className="content-accessory content-end-accessory content-accessory-align-center"
    content={
      <p>
        end
      </p>
    }
    key="end_accessory"
  />
</TableRow>
`;

exports[`Table View Tests should render a selected Item 1`] = `
<TableRow
  isSelected={true}
>
  <TableCell
    className="content-accessory content-accessory-align-center"
    content={
      <p>
        start
      </p>
    }
    key="start_accessory"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 1"
      />
    }
    key="display_1/.0"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 2"
      />
    }
    key="display_2/.1"
  />
  <TableCell
    className="content-display"
    content={
      <ItemDisplay
        iconAlignment="center"
        isDisabled={false}
        isTruncated={false}
        text="Display 3"
      />
    }
    key="display_3/.2"
  />
  <TableCell
    className="content-comment"
    content={
      <ItemComment
        isTruncated={false}
        text="comment"
      />
    }
    key="comment"
  />
  <TableCell
    className="content-accessory content-end-accessory content-accessory-align-center"
    content={
      <p>
        end
      </p>
    }
    key="end_accessory"
  />
</TableRow>
`;

exports[`should render a default Item 1`] = `
<ListItem
  content={
    <ItemView
      accessoryAlignment="alignCenter"
      displays={Array []}
      isTruncated={false}
      layout="oneColumn"
      reserveStartAccessorySpace={false}
      textEmphasis="default"
    />
  }
  hasChevron={false}
  isSelected={false}
/>
`;
